docs/source/oauth2_priv.rst

```diff
@@ -0,0 +1,126 @@
+Privacy-Enhanced OAuth 2.0
+==========================
+
+**Status:** DRAFT
+
+.. contents:: Table of Contents
+    :local:
+    :backlinks: top
+
+Contributors
+------------
+
+* George Aristy (SecureKey Techologies)
+
+Introduction
+------------
+
+Purpose of this document
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+This document describes a reference implementation of OAuth 2.0 with
+unregistered clients communicating and authenticating securely over the
+backchannel with decentralized identifiers and verifiable credentials.
+
+Motivation
+^^^^^^^^^^
+
+There is a desire to leverage existing OAuth 2.0 infrastructure to build a
+privacy-enhanced data sharing solution.
+
+Finalized in 2012, OAuth 2.0 ([RFC6749]_), is an established authorization
+framework well suited to give a piece of software access to protected
+resources with the owner's consent. It was not, however, designed with the
+principles of Privacy by Design ([PRIV-DESIGN]_) in mind.
+
+First published in 2009, the principles of privacy by design became widely
+known after the GDPR adopted them ([GDPR-PRIV]_) and began enforcement in
+2018. We seek to address two key principles of privacy by design that OAuth
+2.0 does not:
+
+**Protect the user's privacy by keeping the solution User-Centric**
+
+* Use the authorization grant mechanism of OAuth 2.0 to keep the user in the
+  locus of control.
+* Use decentralized identifiers (DIDs) ([DID-CORE]_) so the user (and the other
```

Comment:
We will also use Verifiable Credentials to allow the Client and RS to identify some attributes about each other.

Reply:
I am having difficulties integrating a story for verifiable credentials. Can you provide one? For the time being I am assuming that part is out of scope **of this design document** (aka. out of scope of privacy-enhanced authorization with OAuth 2.0) and I am picturing it more as just another form of data that will flow through the pipes.

Reply:
I'm thinking of this more of an exercise in how the user can connect two parties together and have data flowing between them.